src/v/cluster/rm_stm.cc

```diff
@@ -671,6 +671,60 @@ ss::future<> rm_stm::stop() {
     return raft::state_machine::stop();
 }
 
+rm_stm::transaction_info::status_t
+rm_stm::get_tx_status(model::producer_identity pid) {
+    if (_mem_state.preparing.find(pid) != _mem_state.preparing.end()) {
+        return transaction_info::preparing;
+    }
+
+    if (_log_state.prepared.find(pid) != _log_state.prepared.end()) {
+        return transaction_info::prepared;
+    }
+
+    return transaction_info::ongoing;
+}
+
+std::optional<rm_stm::expiration_info>
+rm_stm::get_expiration_info(model::producer_identity pid) {
+    auto it = _mem_state.expiration.find(pid);
+    if (it == _mem_state.expiration.end()) {
+        return std::nullopt;
+    }
+
+    return it->second;
+}
+
+absl::btree_map<model::producer_identity, rm_stm::transaction_info>
+rm_stm::get_transactions() {
+    absl::btree_map<model::producer_identity, rm_stm::transaction_info> ans;
+
+    for (auto& [id, offset] : _mem_state.estimated) {
+        transaction_info tx_info;
+        tx_info.first_offset = offset;
```

Comment:
when redpanda starts writing the first batch of a transaction it estimates its offset and only when the write passes it updates the offset to the exact value; so for a short period of time first_offset maybe incorrect. I don't like to give users a false signal so it's introduce a new status `initiating` for estimated transactions